kerastuner/applications/augment.py

```diff
@@ -0,0 +1,191 @@
+
+import random
+
+# dict of functions that create layers for transforms.
+# Each function takes a factor (0 to 1) for the strength
+# of the transform.
+TRANSFORMS = {
+'translate_x': lambda x: preprocessing.RandomTranslation(x, 0),
+'translate_y': lambda y: preprocessing.RandomTranslation(0, y),
+'rotate': preprocessing.RandomRotation,
+'contrast': preprocessing.RandomContrast,
+}
+
+class HyperAugment(hypermodel.HyperModel):
+    """ Builds HyperModel for image augmentation.
+
+    # Arguments:
+        input_shape: Optional shape tuple, e.g. `(256, 256, 3)`.
+        input_tensor: Optional Keras tensor (i.e. output of
+            `layers.Input()`) to use as image input for the model.
+        rotate: Whether to include rotation in augmentation.
+        translate_x: Whether to include horizontal translation in augmentation.
+        translate_y: Whether to include vertical translation in augmentation.
+        contrast: Whether to include random contrast in augmentation.
+        **kwargs: Additional keyword arguments that apply to all
+            HyperModels. See `kerastuner.HyperModel`.
+    """
+    def __init__(self,
+        input_shape=None,
+        input_tensor=None,
+        rotate=True,
+        translate_x=True,
+        translate_y=True,
+        contrast=True):
+        self.transforms = []
+        if rotate:
+            self.transforms.append('rotate')
+        if translate_x:
+            self.transforms.append('translate_x')
+        if translate_y:
+            self.transforms.append('translate_y')
+        if contrast:
+            self.transforms.append('contrast')
+
+        self.input_shape = input_shape
+        self.input_tensor = input_tensor
+
+    def build(self, hp):
+        raise NotImplemented
+
+class HyperFixedAugment(HyperAugment):
+    """An HyperModel for fixed policy augmentation.
+       A tunable hyperparameter is assigned to each of the
+       transforms chosen, and the transforms are applied
+       sequentially.
+       Only supporting augmentations in Keras preprocessing layers.
+
+    # Arguments:
+        input_shape: Optional shape tuple, e.g. `(256, 256, 3)`.
+        input_tensor: Optional Keras tensor (i.e. output of
+            `layers.Input()`) to use as image input for the model.
+        rotate: Whether to include rotation in augmentation. When
+             set to True, RandomRotation layer will be added to the
+             pool of augmentations to select from.
+        translate_x: Whether to include horizontal translation in augmentation.
+        translate_y: Whether to include vertical translation in augmentation.
+        contrast: Whether to include random contrast in augmentation.
+        **kwargs: Additional keyword arguments that apply to all
+            HyperModels. See `kerastuner.HyperModel`.
+
+    # Search space:
+        'factor_{transform}' for each of the transforms chosen:
+            Float, default range [0.05, 1]. Controls the strength
```

Comment:
Should we allow user to config this?

Reply:
This can be controlled by overriding hyperparameters outside, for example
```
hp = HyperParameters()
hp.Float('factor_translate_x', 0, 0.1, step=0.01)
```
where hp will be used on constructing `tuner`.

It may also be useful to allow controlling the ranges using keyword arguments. Do you think it would be better to 1) add keywords for ranges corresponding to each transform; or 2) allow the keyword for each transform to be boolean, number or a tuple, which overrides the default config if set as number/tuple?

Reply:
Injecting items outside of augment model seems to be confusing, on the order hand, injecting a bunch of kwargs is also errorprone. @omalleyt12, do u have any recommendation here?